pkg/sidecar/pod.go

```diff
@@ -33,8 +35,39 @@ const (
 
 // add a new sidecar container to the given pod, based on the given OpenTelemetryCollector.
 func add(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTelemetryCollector, pod corev1.Pod, attributes []corev1.EnvVar) (corev1.Pod, error) {
-	// add the container
 	volumes := collector.Volumes(cfg, otelcol)
+	volumes[0] = corev1.Volume{
+		Name: naming.ConfigMapVolume(),
+		VolumeSource: corev1.VolumeSource{
+			EmptyDir: &corev1.EmptyDirVolumeSource{},
+		},
+	}
+	volumeMount := corev1.VolumeMount{
+		Name:      naming.ConfigMapVolume(),
+		MountPath: "/conf",
+	}
+
+	otelColCfg, err := reconcile.ReplaceConfig(otelcol)
+	if err != nil {
+		return pod, err
+	}
+	otelColCfgStr := base64.StdEncoding.EncodeToString([]byte(otelColCfg))
```

Comment:
Why the base64 encoding and decoding?  The config should be text, no?

Reply:
The issue is with new line symbols:

```yaml
  initContainers:
  - args:
    - -c
    - echo ${OTEL_CONFIG} > /conf/collector.yaml && cat /conf/collector.yaml
    command:
    - /bin/sh
    env:
    - name: OTEL_CONFIG
      value: |
        receivers:
          otlp:
            protocols:
              grpc:
              http:
        processors:
          memory_limiter:
            check_interval: 1s
            limit_percentage: 75
            spike_limit_percentage: 15
          batch:
            send_batch_size: 10000
            timeout: 10s

        exporters:
          logging:

        service:
          pipelines:
            traces:
              receivers: [otlp]
              processors: []
              exporters: [logging]

```
produces:

```bash
k logs myapp -c otc-container-config-prepper                                                                                                                                                    ploffay@fedora
receivers: otlp: protocols: grpc: http: processors: memory_limiter: check_interval: 1s limit_percentage: 75 spike_limit_percentage: 15 batch: send_batch_size: 10000 timeout: 10s exporters: logging: service: pipelines: traces: receivers: [otlp] processors: [] exporters: [logging]

```

Reply:
Huh, apparently another example of zsh's superiority as its `echo` deals with this just fine.  With bash you can preserve the newlines by quoting the variable:

```sh
echo "${OTEL_CONFIG}" > /conf/collector.yaml && cat /conf/collector.yaml
```

Reply:
yuh, that works. Thanks 👍🏼